hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token/records/TokenBurnRecordBuilder.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.token.records;
+
+import com.hedera.node.app.spi.workflows.record.SingleTransactionRecordBuilder;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.List;
+
+public interface TokenBurnRecordBuilder extends SingleTransactionRecordBuilder {
+
+    @NonNull
+    TokenMintRecordBuilder newTotalSupply(final long newTotalSupply);
```

Comment:
Typo---`TokenMint` instead of `TokenBurn`.

Here if we add a `getNewTotalSupply()` method and implement in `SingleTransactionRecordBuilderImpl` we can then fix the hard-coded `0L` above with the correct total supply.